### PR TITLE
[17.09] builder: fix config leakage on shared parent stage

### PR DESCRIPTION
**Pull request by tonistiigi**
fixes moby/moby/issues/33753

@dnephin 

Signed-off-by: Tonis Tiigi <tonistiigi@gmail.com>

### CODE REVIEW

**Comment by vieux** [comment]
ping @andrewhsu we should take this one

**Comment by vieux** [comment]
LGTM

File: components/engine/integration-cli/docker_cli_build_test.go [hunk 1]
```diff
@@ -6234,6 +6234,28 @@ func (s *DockerSuite) TestBuildCopyFromResetScratch(c *check.C) {
 	c.Assert(strings.TrimSpace(res), checker.Equals, "")
 }
 
+func (s *DockerSuite) TestBuildMultiStageParentConfig(c *check.C) {
```
Comment (thaJeztah) [inline components/engine/integration-cli/docker_cli_build_test.go]: Should we add this test to the https://github.com/moby/moby repository as well?

Comment (dnephin) [reply components/engine/integration-cli/docker_cli_build_test.go]: hopefully with the refactor we should be able to cover this with a unit test in moby/moby

**Pull request merged**